src/toil/test/server/wes_cwl_runner.py

```diff
@@ -0,0 +1,165 @@
+                loc = file.get("location")
+
+                # Check directory
+                if file.get("class") == "Directory":
+                    if file.get("listing"):
+                        get_deps(file.get("listing"))
+                    else:
+                        # no listing, so import all files in the directory
+                        for folder, _, sub_files in os.walk(loc):
+                            for sub_file in sub_files:
+                                deps.append(os.path.join(folder, sub_file))
+                else:
+                    deps.append(file.get("location"))
+
+                # check secondaryFiles
+                if "secondaryFiles" in file:
+                    get_deps(file.get("secondaryFiles"))
+
+    get_deps(json_result.get("secondaryFiles", []))
+    return deps
+
+
+def submit_run(client: WESClient,
+               cwl_file: str,
+               input_file: Optional[str] = None,
+               engine_options: Optional[List[str]] = None) -> str:
+    # First, get the list of files to attach to this workflow
+    attachments = get_deps_from_cwltool(cwl_file, input_file)
+
+    if input_file:
+        attachments.extend(get_deps_from_cwltool(cwl_file, input_file, input_deps=True))
+
+    # logging.warning(f"Files to import: {attachments}")
+
+    run_result: Dict[str, Any] = client.run(cwl_file, input_file,
+                                            attachments=attachments,
+                                            engine_options=engine_options)
+    return run_result.get("run_id", None)
+
+
+def poll_run(client: WESClient, run_id: str) -> bool:
+    """ Return True if the given workflow run is in a finished state."""
+    status_result = client.get_run_status(run_id)
+    state = status_result.get("state")
+
+    return state in ("COMPLETE", "CANCELING", "CANCELED", "EXECUTOR_ERROR", "SYSTEM_ERROR")
+
+
+def print_logs_and_exit(client: WESClient, run_id: str) -> None:
+    # cwltest checks for stdout, stderr, and return code
+    data = client.get_run_log(run_id)
+
+    outputs = json.dumps(data.get("outputs", {}), indent=4)
+    exit_code = data.get("run_log", {}).get("exit_code", 1)
+
+    sys.stdout.write(outputs)
+    sys.exit(exit_code)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    # first two positional arguments are passed in by cwltest
+    parser.add_argument("cwl_file", type=str)
+    parser.add_argument("input_file", type=str, nargs="?", default=None)
+    # a few arguments useful for the WES runner
+    parser.add_argument("--wes_endpoint", default="http://localhost:8080",
+                        help="The http(s) URL of the WES server.  (default: %(default)s)")
+    parser.add_argument("--wes_user", default=None,
+                        help="Username to use for basic authentication to the WES server.")
+    parser.add_argument("--wes_password", default=None,
+                        help="Password to use for basic authentication to the WES server.")
```

Comment:
It would be great if this one in particular could come from an environment variable.